crates/oxc_linter/src/rules/typescript/consistent_type_definitions.rs

```diff
@@ -0,0 +1,214 @@
+use oxc_ast::{
+    ast::{ExportDefaultDeclarationKind, TSType},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("typescript-eslint(consistent-type-definitions):")]
+#[diagnostic(severity(warning), help("Use an `{0}` instead of a `{1}`"))]
+struct ConsistentTypeDefinitionsDiagnostic(
+    &'static str,
+    &'static str,
+    #[label("Use an `{0}` instead of a `{1}`")] pub Span,
+);
+
+#[derive(Debug, Default, Clone)]
+pub struct ConsistentTypeDefinitions {
+    config: ConsistentTypeDefinitionsConfig,
+}
+
+#[derive(Debug, Default, Clone, Copy, Eq, PartialEq)]
+enum ConsistentTypeDefinitionsConfig {
+    #[default]
+    Interface,
+    Type,
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    ///
+    /// ### Why is this bad?
+    ///
+    ///
+    /// ### Example
+    /// ```javascript
+    /// ```
+    ConsistentTypeDefinitions,
+    style
+);
+
+impl Rule for ConsistentTypeDefinitions {
+    fn from_configuration(value: serde_json::Value) -> Self {
+        let config = value.get(0).and_then(serde_json::Value::as_str).map_or_else(
+            ConsistentTypeDefinitionsConfig::default,
+            |value| match value {
+                "type" => ConsistentTypeDefinitionsConfig::Type,
+                _ => ConsistentTypeDefinitionsConfig::Interface,
+            },
+        );
+        Self { config }
+    }
+
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        match node.kind() {
+            AstKind::TSTypeAliasDeclaration(typ) => match &typ.type_annotation {
+                TSType::TSTypeLiteral(lit)
+                    if self.config == ConsistentTypeDefinitionsConfig::Interface =>
+                {
+                    ctx.diagnostic(ConsistentTypeDefinitionsDiagnostic("interface", "type", lit.span))
```

Comment:
There is [fixer](https://github.com/typescript-eslint/typescript-eslint/blob/fde44275117b3a7bbdcaaf267c73986b6637c5f9/packages/eslint-plugin/src/rules/consistent-type-definitions.ts#L54-L82) for this rule, are you having interest to achieving it too :)